src/SLCore/ISLCoreHandler.cs

```diff
@@ -20,116 +20,81 @@
+        ISloopRestartFailedNotificationService notificationService,
+        IThreadHandling threadHandling)
+        : this(slCoreInstanceHandler, notificationService, 3, threadHandling)
     {
-        this.aliveConnectionTracker = aliveConnectionTracker;
-        this.activeConfigScopeTracker = activeConfigScopeTracker;
-        this.slCoreHandleFactory = slCoreHandleFactory;
-        this.threadHandling = threadHandling;
-        this.logger = logger;
     }
 
-    public async Task StartInstanceAsync()
+    internal /* for testing */ SLCoreHandler(ISLCoreInstanceHandler slCoreInstanceHandler,
+        ISloopRestartFailedNotificationService notificationService,
+        int maxAutoStartsBeforeManual,
+        IThreadHandling threadHandling)
     {
-        threadHandling.ThrowIfOnUIThread();
-
-        if (disposed)
+        if (maxAutoStartsBeforeManual <= 0)
         {
-            throw new ObjectDisposedException(nameof(SLCoreHandler));
+            throw new ArgumentOutOfRangeException(nameof(maxAutoStartsBeforeManual));
         }
 
-        if (currentHandle is not null)
-        {
-            throw new InvalidOperationException(SLCoreStrings.SLCoreHandler_InstanceAlreadyRunning);
-        }
-        
-        if (!TryCreateInstance())
-        {
-            return;
-        }
-
-        await LaunchInstanceAsync();
-    }
-
-    private bool TryCreateInstance()
-    {
-        CurrentStartNumber++;
-        try
-        {
-            logger.WriteLine(SLCoreStrings.SLCoreHandler_CreatingInstance);
-            currentHandle = slCoreHandleFactory.CreateInstance();
-        }
-        catch (Exception e)
-        {
-            logger.WriteLine(SLCoreStrings.SLCoreHandler_CreatingInstanceError);
-            logger.LogVerbose(e.ToString());
-            return false;
-        }
-
-        return true;
+        this.notificationService = notificationService;
+        this.slCoreInstanceHandler = slCoreInstanceHandler;
+        this.threadHandling = threadHandling;
+        this.maxAutoStartsBeforeManual = maxAutoStartsBeforeManual;
     }
 
-    private async Task LaunchInstanceAsync()
+    public void EnableSloop()
     {
-        try
-        {
-            logger.WriteLine(SLCoreStrings.SLCoreHandler_StartingInstance);
-            await currentHandle.InitializeAsync();
-            await currentHandle.ShutdownTask;
-        }
-        catch (Exception e)
+        if (disposed)
         {
-            logger.WriteLine(SLCoreStrings.SLCoreHandler_StartingInstanceError);
-            logger.LogVerbose(e.ToString());
+            throw new ObjectDisposedException(nameof(SLCoreHandler));
         }
-        finally
+        
+        threadHandling.RunOnBackgroundThread(async () =>
         {
-            logger.WriteLine(SLCoreStrings.SLCoreHandler_InstanceDied);
-            HandleInstanceDeath();
-        }
+            while (!disposed && slCoreInstanceHandler.CurrentStartNumber - initiatedStartAtCount < maxAutoStartsBeforeManual)
```

Comment:
Should be <= I believe run number should be "restartnumber + 1"

Reply:
the initial start is not counted as a restart, there are tests for that as well